alerts/kubelet.libsonnet

```
@@ -37,6 +40,24 @@ local utils = import '../lib/utils.libsonnet';
             'for': '15m',
             alert: 'KubeNodeNotReady',
           },
+          {
+            alert: 'KubeNodePressure',
+            expr: |||
+              kube_node_status_condition{%(kubeStateMetricsSelector)s,condition=!"Ready",status="true"} == 1
```

When using `condition!="Ready"`, it opens the query up to many more conditions which are likely not pressure related. Is this the intention?

From a quick test, I see at least 23 different conditions including:

- `VMEventScheduled`
- `KernelDeadlock`
- `SysctlChanged`

To name just a few.

Maybe we should enumerate the full list of conditions this alert is designed for? From the [docs](https://kubernetes.io/docs/concepts/scheduling-eviction/node-pressure-eviction/#node-conditions):

- `MemoryPressure`
- `DiskPressure`
- `PIDPressure`

So maybe:

```suggestion
              kube_node_status_condition{%(kubeStateMetricsSelector)s,condition=~"(MemoryPressure|DiskPressure|PIDPressure)",status="true"} == 1
```

wdyt? Am I missing any?

---

hmmm good catch, I missed that.  In my cluster I spotted the label `condition=NetworkUnavailable` and thought that would be handy to include, but didn't consider that there were even more possible values.  But I suppose NetworkUnavailable isn't a pressure condition.  So do we want to limit this alert to just pressure conditions?

---

I think it makes sense to target at least this PR to be pressure conditions only, as that's mostly what we've explored and discussed.

Happy to consider further node alerts in future PRs (or indeed to expand on these new alert rules if necessary), but I'd rather do some more research on the extent and impact of the other non-pressure conditions - and whether custom conditions are possible.